core/src/main/java/com/linecorp/armeria/client/AggregatedHttpRequestHandler.java

```diff
@@ -70,20 +72,29 @@ private void apply0(@Nullable AggregatedHttpRequest request, @Nullable Throwable
             return;
         }
 
-        HttpData content = request.content();
+        aReq = request;
+        if (state() != State.NEEDS_100_CONTINUE) {
```

Comment:
What do you think of using `HttpRequestSubscriber` if `Expect: 100-continue` header is set for simplicity? I don't think we need to implement support for `Expect: 100-continue` for all request types. 
```java
if (!ctx.exchangeType().isRequestStreaming() && shouldExpect100Continue()) {
   final AggregatedHttpRequestHandler reqHandler = new AggregatedHttpRequestHandler(
   ...
}
```
- `Expect: 100-continue` is usually used when a request body is large.
- It is not common to use 100-continue with WebSocket.

Reply:
As 100-continue is not commonly used for REST API, I wanted to keep the optimal code path for most requests. 

That said, the impact on performance would be trivial with the current changes. I agree with your idea. It makes sense.

Reply:
I believe we should also support for  request streaming type because a user wants to send a unary POST request with `ExchangeType.BIDI_STREAMING` type. Let me fix this in the follow-up PR.